stdlib/statistics.pyi

```diff
@@ -70,3 +77,11 @@ if sys.version_info >= (3, 8):
         def __rsub__(self, x2: Union[float, NormalDist]) -> NormalDist: ...
         __rmul__ = __mul__
         def __hash__(self) -> int: ...
+
+if sys.version_info >= (3, 10):
+    def correlation(__x: Sequence[_Number], __y: Sequence[_Number]) -> float: ...
```

Comment:
correlation and linear_regression also throw when mixing different types, but covariance doesn't somehow, at least for the example I tried. That means at least the first two should also use `_NumberT`.